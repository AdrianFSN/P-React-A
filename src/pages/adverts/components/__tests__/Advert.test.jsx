import { Provider } from "react-redux";
import Advert from "../Advert";
import { render } from "@testing-library/react";

describe("Advert component", () => {
  const state = {
    ads: {
      data: [
        {
          id: "",
          name: "",
          price: 0,
          sale: false,
          tags: [],
        },
      ],
    },
  };
  const store = {
    dispatch: () => {},
    getState: () => state,
    subscribe: () => {},
  };
  const advert = {
    id: "1",
    name: "Advert Name",
    price: 25,
    sale: true,
    tags: ["motor", "mobile"],
    photo: "ad photo",
  };

  const renderComponent = () =>
    render(
      <Provider store={store}>
        <Advert
          id={advert.id}
          name={advert.name}
          price={advert.price}
          sale={advert.sale}
          tags={advert.tags}
          photo={advert.photo}
        />
      </Provider>
    );
  it("should take a snapshot for Advert component", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
