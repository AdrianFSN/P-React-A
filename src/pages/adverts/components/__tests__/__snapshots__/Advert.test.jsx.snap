// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advert component should take a snapshot for Advert component 1`] = `
<div>
  <section
    class="single-advert"
  >
    <div
      class="offer-type-circle on-sale"
    >
      On sale
    </div>
    <ul>
      <li>
        <h2>
          Advert Name
        </h2>
      </li>
      <li>
        Price: 
        25
         â‚¬
      </li>
      <li>
        Category: 
        motor | mobile
      </li>
    </ul>
  </section>
</div>
`;
